common/aws/test/client_test.go

```diff
@@ -0,0 +1,176 @@
+package test
+
+import (
+	"context"
+	"github.com/Layr-Labs/eigenda/common"
+	"github.com/Layr-Labs/eigenda/common/aws"
+	"github.com/Layr-Labs/eigenda/common/aws/s3"
+	"github.com/Layr-Labs/eigenda/common/mock"
+	tu "github.com/Layr-Labs/eigenda/common/testutils"
+	"github.com/Layr-Labs/eigenda/inabox/deploy"
+	"github.com/ory/dockertest/v3"
+	"github.com/stretchr/testify/assert"
+	"math/rand"
+	"os"
+	"testing"
+)
+
+var (
+	dockertestPool     *dockertest.Pool
+	dockertestResource *dockertest.Resource
+)
+
+const (
+	localstackPort = "4570"
```

Comment:
consider using 0 to let the os assign a random port for each instantiation of localstack. With this you can then run all tests in parallel by using `t.parallel()` at the beginning of your tests: https://pkg.go.dev/testing#T.Parallel

Reply:
This is the only port number I can get to work. When this test runs in the github CI job, there is already a local stack instance with this port. In that environment, my test harness doesn't start a new instance, it just uses the existing one. Although it was easy to get the test running locally with a port of my choosing, I was unable to get that working in the CI pipeline.

Reply:
I see.... that's kinda weird. So our github workflow starts a global localstack instance before running the tests? Feel like the localstack should be controlled by the tests and started by them, but maybe there's something I'm missing..